File: 2324A/NewBB mocks/mock2/integerlists_Amanda_deque.cpp
/*
*OpenKattis
*integerlists
*Code by: Amanda Lim
*/

#include <bits/stdc++.h>
using namespace std;

int main ()
{
    ios::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);
    
    int T;
    cin >> T;
    for (int t = 0; t < T; t++) {
        string BAPC;
        cin >> BAPC;

        int N;
        cin >> N;

        char tapon;
        cin >> tapon;

        deque <int> list_of_int (N);
        for (int n = 0; n < N; n++)
            cin >> list_of_int[n] >> tapon;
        if (!N)
            cin >> tapon;

        bool error = false;
        bool reverse = false;
        for (int i = 0; i < (int) BAPC.size(); i++) {
            if (BAPC[i] == 'R')
                reverse = !reverse;
            else {
                if (list_of_int.empty()) {
                    error = true;
                    break;
                }

                if (!reverse)
                    list_of_int.pop_front();
                else
                    list_of_int.pop_back();
            }
        }

        if (error)
            cout << "error" << endl;
        else {
            cout << '[';
            if (!reverse)
                for (int i = 0; i < (int) list_of_int.size(); i++) {
                    cout << list_of_int[i];
                    if (i < ((int) list_of_int.size() - 1))
                        cout << ',';
                }
            else
                for (int i = list_of_int.size() - 1; i >= 0; i--) {
                    cout << list_of_int[i];
                    if (i > 0)
                        cout << ',';
                }
            cout << ']' << endl;
        }
    }

    return 0;
}

File: 2324A/NewBB mocks/mock2/integerlists_Amanda_deque.py
'''
OpenKattis
integerlists
Code by: Amanda Lim
'''

from collections import deque

T = int(input())
for t in range(T):
    BAPC = input().strip()

    N = int(input())
    list_of_int = input().strip()
    list_of_int = list_of_int[1:-1]
    
    list_of_int = deque(list_of_int.split(','))
    if (N == 0):
        list_of_int.clear()

    reverse = False
    error = False
    for i in BAPC:
        if i == 'R':
            reverse = not reverse
        else:
            if bool(list_of_int) == False:
                error = True
                break
            if reverse:
                list_of_int.pop()
            else:
                list_of_int.popleft()
    
    if error:
        print("error")
    else:
        print('[', end="")
        if reverse:
            for i in range(len(list_of_int) - 1, -1,-1):
                print(list_of_int[i], end="")
                if (i > 0):
                    print(',', end="")
        else:
            for i in range(0,len(list_of_int)):
                print(list_of_int[i], end="")
                if (i < len(list_of_int) - 1):
                    print(',', end="")
        print(']')